Common/File/FileUtil.cpp

```diff
@@ -519,8 +519,8 @@ bool CreateDir(const Path &path) {
 		// Convert it to a "CreateDirIn" call, if possible, since that's
 		// what we can do with the storage API.
 		AndroidContentURI uri(path.ToString());
-		std::string newDirName = uri.GetLastPart();
 		if (uri.NavigateUp()) {
+			std::string newDirName = uri.GetLastPart();
```

Comment:
This change broke directory creation, because uri.NavigateUp actually changes the URI object, so the order changes.